src/components/pages/lessons/overlay/go-pro-cta-overlay.tsx

```diff
@@ -2,30 +2,170 @@ import React, {FunctionComponent} from 'react'
+
+  const pppCouponAvailable =
+    !isEmpty(countryName) && !isEmpty(countryCode) && !isEmpty(parityCoupon)
+  const pppCouponEligible = quantity === 1
+
+  // handlers
+  const onApplyParityCoupon = () => {
+    send('APPLY_PPP_COUPON')
+  }
+
+  const onDismissParityCoupon = () => {
+    send('REMOVE_PPP_COUPON')
+  }
+
+  const onClickCheckout = async () => {
+    if (!priceId) return
+    if (!emailForCheckout) return
+
+    const account = first<StripeAccount>(get(viewer, 'accounts'))
+    await track('checkout: selected plan', {
+      priceId: priceId,
+    })
+
+    const emailRequiresAProCheck =
+      !isEmpty(emailForCheckout) && emailForCheckout !== viewer?.email
+
+    if (emailRequiresAProCheck) {
+      const {hasProAccess} = await axios
+        .post(`/api/users/check-pro-status`, {
+          email: emailForCheckout,
+        })
+        .then(({data}) => data)
+
+      if (hasProAccess) {
+        const message = `You've already got a pro account at ${emailForCheckout}. Please sign in.`
+
+        toast.error(message, {
+          duration: 6000,
+          icon: '❌',
+        })
+
+        track('cta overlay checkout: existing pro account found', {
+          email: emailForCheckout,
+        })
+
+        // email is already associated with a pro account, return early instead
+        // of sending the user to a Stripe Checkout Session.
+        return
+      }
+    }
+
+    if (emailIsValid(emailForCheckout)) {
+      await track('checkout: valid email present', {
+        priceId: priceId,
+      })
+      await track('checkout: redirect to stripe', {
+        priceId,
+      })
+
+      stripeCheckoutRedirect({
+        priceId,
+        email: emailForCheckout,
+        stripeCustomerId: account?.stripe_customer_id,
+        authToken,
+        quantity,
+        coupon: state.context.couponToApply?.couponCode,
+      })
+    } else {
+      // we don't have a valid email for the checkout
+      await track('checkout: unable to proceed, no valid email', {
+        email: emailForCheckout,
+      })
+    }
+  }
+
   return (
     <div className="flex flex-col items-center p-4">
-      <h2 className="sm:text-2xl text-xl leading-tighter tracking-tight font-light text-center">
-        This lesson is an{' '}
-        <strong className="font-bold">egghead member exclusive</strong>.
+      <h2 className="sm:text-lg text-sm uppercase leading-tighter tracking-tight font-light text-center">
+        This lesson is for members only
       </h2>
-      <Link href="/pricing">
-        <a
-          onClick={() => {
-            track('clicked join cta on blocked lesson', {
-              lesson: lesson.slug,
-            })
+      <h1 className="sm:text-2xl text-xl leading-tighter tracking-tight font-light text-center">
+        Ready to take your career to the next level?
+      </h1>
+      <PlanPrice pricesLoading={pricesLoading} plan={currentPlan} />
+      {isEmpty(viewer) && (
+        <input
```

Comment:
We only display the Email Input if there isn't a viewer (i.e. the visitor isn't signed in)